core/server/data/migrations/versions/4.9/05-fix-missed-mobiledoc-url-transforms.js

```diff
@@ -0,0 +1,73 @@
+const logging = require('@tryghost/logging');
+const semver = require('semver');
+const urlUtils = require('../../../../../shared/url-utils');
+const htmlToPlaintext = require('../../../../../shared/html-to-plaintext');
+const mobiledocLib = require('../../../../lib/mobiledoc');
+const {createTransactionalMigration} = require('../../utils');
+
+// in Ghost versions 4.6.1-4.8.4 the 4.0 migration that transfored URLs had a bug
+// that meant urls inside cards in mobiledoc content was not being transformed
+//
+// if the migrations table indicates an upgrade was made from 3.x to 4.6-4.8 then
+// we'll re-run the transforms against post.mobiledoc and re-generate the html
+// and plaintext contents
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const versionUsedFor40Migration = await knex('migrations')
+            .where({
+                name: '18-transform-urls-absolute-to-transform-ready.js',
+                version: '4.0'
+            })
+            .select('currentVersion')
+            .first();
+
+        // migrations table at time of writing only stores major.minor so we need to coerce
+        // and be liberal with the bad version range
+        const currentVersion = semver.coerce(versionUsedFor40Migration.currentVersion);
+        const badVersionRange = '4.6.0 - 4.8.4';
+
+        if (!semver.satisfies(currentVersion, badVersionRange)) {
+            logging.info('Skipping transform of mobiledoc URLs - original transform was good');
+            return;
+        }
+
+        logging.info('Transforming all internal URLs in posts.{mobiledoc,html,plaintex} to transform-ready');
+
+        await knex.transaction(async (trx) => {
+            const postIdRows = await knex('posts')
+                .transacting(trx)
+                .forUpdate()
+                .select('id');
+
+            for (const postIdRow of postIdRows) {
+                const {id} = postIdRow;
+                const [post] = await knex('posts')
+                    .transacting(trx)
+                    .where({id})
+                    .select([
+                        'mobiledoc'
+                    ]);
+
+                const mobiledoc = urlUtils.mobiledocToTransformReady(post.mobiledoc, {cardTransformers: mobiledocLib.cards});
```

Comment:
Would we be able to wrap this in try-catches like we did for a similar [4.0 migration](https://github.com/TryGhost/Ghost/blob/fec534f83374a4c4d0ddafb84d487b71eb96d2ba/core/server/data/migrations/versions/4.0/23-regenerate-posts-html.js#L29-L37)? It saves us from crashing out if the mobiledoc is in a bad format